lib/services/nodes-api-service.js

```diff
@@ -235,8 +235,16 @@ function nodeApiServiceFactory(
      * @returns {Promise}
      */
 
-    NodeApiService.prototype.getAllNodes = function(query) {
-        return waterline.nodes.find(query).populate('obms');
+    NodeApiService.prototype.getAllNodes = function(query, options) {
+        options = options || {};
+        return Promise.try(function() {
+            var query = waterline.nodes.find(query);
+
+            if (options.skip) query.skip(options.skip);
+            if (options.limit) query.limit(options.limit);
+
+            return query.populate('obms');
```

Comment:
I notice the function https://github.com/RackHD/on-http/blob/master/lib/services/nodes-api-service.js#L238  it could be refined here or there